cmd/utils/flags.go

```diff
@@ -447,6 +451,33 @@ func MakeEthConfig(clientID, version string, ctx *cli.Context) *eth.Config {
 		SolcPath:                ctx.GlobalString(SolcPathFlag.Name),
 		AutoDAG:                 ctx.GlobalBool(AutoDAGFlag.Name) || ctx.GlobalBool(MiningEnabledFlag.Name),
 	}
+
+	if ctx.GlobalBool(DevModeFlag.Name) {
+		if !ctx.IsSet(VMDebugFlag.Name) {
+			cfg.VmDebug = true
+		}
+		if !ctx.IsSet(MaxPeersFlag.Name) {
+			cfg.MaxPeers = 0
+		}
+		if !ctx.IsSet(GasPriceFlag.Name) {
+			cfg.GasPrice = new(big.Int)
+		}
+		if !ctx.IsSet(ListenPortFlag.Name) {
+			cfg.Port = "0" // auto port
+		}
+		if !ctx.IsSet(WhisperEnabledFlag.Name) {
+			cfg.Shh = true
+		}
+		if !ctx.IsSet(DataDirFlag.Name) {
+			cfg.DataDir = os.TempDir() + "/ethereum_dev_mode"
```

Comment:
Actually, I think getting it wiped is a good thing. If you corrupt something, then you can start with a clean slate without having to worry about how to clean the mess up, and additionally the bigger benefit is that it forces you to develop stateless code (i.e. that does not depend on some state that you managed to cook up/arrive to after some convoluted procedure, but rather one which can be deployed anytime without worrying about previous state).